images/kubernetes/src/cockpit-kube-launch/main.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package main
+
+import (
+	"cockpit-kube-auth/helpers"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+var OAUTH_CLIENT_ID = "cockpit-kube-client"
+
+func writeFile(filename string, data string) {
+	f, err := os.Create(filename)
+	if err != nil {
+		log.Fatalf("Could't open file %s: %v", filename, err)
+	}
+	_, wErr := f.WriteString(data)
+	if wErr != nil {
+		log.Fatalf("Could't write file %s: %v", filename, wErr)
+	}
+	f.Close()
+}
+
+func setupCertificates() {
+	// Ensure path exits
+	dErr := os.MkdirAll("/etc/cockpit/ws-certs.d", os.ModeDir|0775)
+	if dErr != nil {
+		log.Fatalf("Couldn't create certificate directory %s", dErr)
+	}
+
+	// Setting ownership on the certificate may fail
+	// so execute ensure and the check results without
+	exec.Command("/usr/sbin/remotectl", "certificate", "--ensure").CombinedOutput()
+	out, rErr := exec.Command("/usr/sbin/remotectl", "certificate").CombinedOutput()
+	if rErr != nil {
+		log.Fatalf("Failed to generate certificates %s %s", rErr, out)
+	}
+}
+
+func haveOpenShiftEndpoint() (bool, error) {
+	var isOpenShift bool = false
+
+	creds, err := helpers.NewCredentialsForSystem()
+	if err != nil {
+		return isOpenShift, err
+	}
+
+	client := helpers.NewClient()
+	resp, e := client.DoRequest("GET", "oapi", "", creds, nil)
+	if e != nil {
+		return isOpenShift, e
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode == 200 {
+		isOpenShift = true
+	}
+
+	return isOpenShift, nil
+}
+
+func main() {
+
+	var args []string = nil
+	var isOpenShift bool = false
+
+	if os.Getenv("COCKPIT_KUBE_INSECURE") != "true" {
+		setupCertificates()
+		args = []string{
+			"--no-tls",
+		}
+	}
+
+	filename := os.Args[0]
+	filedirectory := filepath.Dir(filename)
+	location, e := filepath.Abs(filedirectory)
+	if e != nil {
+		log.Fatalf("Could't find location %v", e)
+	}
+
+	path := path.Join(location, "cockpit-kube-auth")
+
+	title := "Cockpit Kubernetes"
+	id := "cockpit-kubernetes"
```

Comment:
Can we build alternate os-release and override.json with the titles (for Openshift, Kubernetes, AEP, etc...) and move them into place?